City-of-Helsinki/drupal-helfi-platform-config
# UHF-4078: Add content lock module dependency and configuration

## How to install
- Make sure your instance is up and running on latest dev branch.
- Update the Helfi Platform config
   - `composer require drupal/helfi_platform_config:dev-UHF-4078_content_lock`
- Run `make drush-updb drush-cr drush-uli`

## How to test
- Go to `/admin/content/locked-content` and make sure you see a listing of locked content (nodes, tpr-services and tpr-units). There shouldn't be anything locked.
- Now to and try to edit any node, tpr-service or tpr-unit and you should get a warning that the entity is not locked for editing and others can't edit it unless they break the lock.
- Now go back to `/admin/content/locked-content` and you should notice that all the language versions of the entity you edited as locked. This might need a cache clear for the tpr-entities at last.
- You should have the same locking functionality for taxonomy terms and paragraph library items. These however are not listed on the page since I thought that this is probably not necessary at this point at least.
- Try breaking the lock (adminsitrator-role people can do that, others can't). You can break the locks from the listing page or by going to the content and pressing the "Break the lock" link on the warning that is telling you that the content is locked.
- Make sure there is translations in place for the new functionality.

Also check this PR:
https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/83
